client/add/add.ctrl.js

```diff
@@ -123,11 +128,16 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
-      .tap(box.onscroll)
+      .filter(line => boxCounts[line.box]++ < BOX_PAGE_SIZE)
```

Comment:
Ah right, I wasn't 100% sure about what it was meant to be doing. That makes sense.

> (Also, I'm not really a big fan of having a filter function with side-effects, but it's not a PR-blocker.)

I mean, the side-effect is purely for the filter itself, boxCounts only exists for this exact function, nothing else. So it isn't **really** a side-effect.